code/datums/spells/mime.dm

```diff
@@ -99,17 +99,27 @@
 	action_icon_state = "fingergun"
 	action_background_icon_state = "bg_mime"
 	var/gun = /obj/item/gun/projectile/revolver/fingergun
+	var/obj/item/gun/projectile/revolver/fingergun/current_gun
 
 /obj/effect/proc_holder/spell/mime/fingergun/cast(list/targets, mob/user = usr)
 	for(var/mob/living/carbon/human/C in targets)
-		if(!istype(C.get_active_hand(), gun) && !istype(C.get_inactive_hand(), gun))
+		if(!current_gun)
 			to_chat(user, "<span class='notice'>You draw your fingers!</span>")
+			current_gun = new gun(user, src)
 			C.drop_item()
-			C.put_in_hands(new gun)
+			C.put_in_hands(current_gun)
+			RegisterSignal(C, COMSIG_MOB_WILLINGLY_DROP, PROC_REF(holster_hand))
 		else
-			to_chat(user, "<span class='notice'>Holster your fingers first.</span>")
+			holster_hand(user, TRUE)
 			revert_cast(user)
 
+/obj/effect/proc_holder/spell/mime/fingergun/proc/holster_hand(atom/target, any=FALSE)
+	SIGNAL_HANDLER
+	if(!current_gun || !any && action.owner.get_active_hand() != current_gun)
+		return
+	to_chat(action.owner, "<span class='notice'>You holster your fingers. Another time.</span>")
```

Comment:
```suggestion
	to_chat(action.owner, "<span class='notice'>You holster your fingers. Another time perhaps...</span>")
```
Me thinks this is like a trailing off thought in the players internal dialogue so you should support that with `...`